services/github/github-package-registry-downloads.service.js

```diff
@@ -0,0 +1,295 @@
+'use strict'
+
+const gql = require('graphql-tag')
+const Joi = require('@hapi/joi')
+const { downloadCount } = require('../color-formatters')
+const { metric } = require('../text-formatters')
+const { nonNegativeInteger } = require('../validators')
+const { GithubAuthV4Service } = require('./github-auth-service')
+const { documentation, transformErrors } = require('./github-helpers')
+const { NotFound } = require('..')
+
+// https://developer.github.com/v4/object/registrypackagestatistics/
+// https://developer.github.com/v4/object/registrypackageversionstatistics/
+const packageStatistics = Joi.object({
+  statistics: Joi.object({
+    downloadsToday: nonNegativeInteger,
+    downloadsThisWeek: nonNegativeInteger,
+    downloadsThisMonth: nonNegativeInteger,
+    downloadsThisYear: nonNegativeInteger,
+    downloadsTotalCount: nonNegativeInteger,
+  }).required(),
+})
+
+const totalDownloadsSchema = Joi.object({
+  data: Joi.object({
+    repository: Joi.object({
+      registryPackages: Joi.object({
+        nodes: Joi.array()
+          .items(packageStatistics)
+          .min(0)
+          .required(),
+      }).required(),
+    }).required(),
+  }).required(),
+}).required()
+
+const versionDownloadsSchema = Joi.object({
+  data: Joi.object({
+    repository: Joi.object({
+      registryPackages: Joi.object({
+        nodes: Joi.array()
+          .items(
+            Joi.object({
+              version: packageStatistics.allow(null),
+            })
+          )
+          .min(0)
+          .required(),
+      }).required(),
+    }).required(),
+  }).required(),
+}).required()
+
+const latestVersionDownloadsSchema = Joi.object({
+  data: Joi.object({
+    repository: Joi.object({
+      registryPackages: Joi.object({
+        nodes: Joi.array()
+          .items(
+            Joi.object({
+              latestVersion: packageStatistics,
+            })
+          )
+          .min(0)
+          .required(),
+      }).required(),
+    }).required(),
+  }).required(),
+}).required()
+
+const intervalMap = {
+  today: {
+    transform: statistics => statistics.downloadsToday,
+    messageSuffix: '/today',
+  },
+  week: {
+    transform: statistics => statistics.downloadsThisWeek,
+    messageSuffix: '/week',
+  },
+  month: {
+    transform: statistics => statistics.downloadsThisMonth,
+    messageSuffix: '/month',
+  },
+  year: {
+    transform: statistics => statistics.downloadsThisYear,
+    messageSuffix: '/year',
+  },
+  total: {
+    transform: statistics => statistics.downloadsTotalCount,
+    messageSuffix: '',
+  },
+}
+
+module.exports = class GithubPackageRegistryDownloads extends GithubAuthV4Service {
+  static get category() {
+    return 'downloads'
+  }
+
+  static get route() {
+    return {
+      base: 'github/packages/downloads',
+      pattern:
+        ':user/:repo/:interval(today|week|month|year|total)/:packageName/:version?',
```

Comment:
Based on what I've seen, all packages have to be associated with a specific repository, even though GitHub does now include a `Packages` menu item at the user/org level that catalogs all the pacakges.